test/src/com/xilinx/rapidwright/eco/TestECOTools.java

```diff
@@ -1118,4 +1119,66 @@ public void testDiscussion1245(String device) {
 
         Assertions.assertEquals("[IN SLICE_X148Y1.B2]", PartialRouter.getUnroutedPins(test).toString());
     }
+
+    @Test
+    public void testConnectNetWithoutSource() {
+        // Also taken from discussion:
+        // https://github.com/Xilinx/RapidWright/discussions/1245#discussioncomment-14044695
+
+        Design design = new Design("clock_gating_test", "vp1202");
+        EDIFCell top = design.getTopEDIFCell();
+
+        Net clk = design.createNet("clk");
+        clk.getLogicalNet().createPortInst(top.createPort("clk", EDIFDirection.INPUT, 1));
+
+        Net gate = design.createNet("gate");
+        gate.getLogicalNet().createPortInst(top.createPort("gate", EDIFDirection.INPUT, 1));
+
+        // TODO - Instantiate IBUFs after primitives fix
+//        Cell clkIBUF = design.createAndPlaceCell(top, "clkIBUF", Unisim.IBUF, "IOB_X29Y0/IOB_M");
+//        ECOTools.connectNet(design, clkIBUF, "I", clk);
+//        
+//        Cell gateIBUF = design.createAndPlaceCell(top, "gateIBUF", Unisim.IBUF, "IOB_X29Y1/IOB_M");
+//        ECOTools.connectNet(design, gateIBUF, "I", gate);
+//
+//        Net clkBUF = design.createNet("clkBUF");
+//        ECOTools.connectNet(design, clkIBUF, "O", clkBUF);
+//        Net gateBUF = design.createNet("gateBUF");
+//        ECOTools.connectNet(design, gateIBUF, "O", gateBUF);
```

Comment:
Are we expecting a primitives fix imminently? This code seems to attach sources to these nets, which sounds like it could defeat the purpose of this test?